src/wasm/wasm-type.cpp

```diff
@@ -2213,9 +2248,11 @@ struct TypeBuilder::Impl {
       // value.
       info = std::make_unique<HeapTypeInfo>(Signature());
       set(Signature());
+      initialized = false;
```

Comment:
This change looks unrelated... but is it?

Reply:
No, this is just a drive by fix. The `initialized` field is checked at `build()` time to make sure that there are no user-uninitialized entries, but we were accidentally setting it to `true` by default because we used `set` to initialize the entries to their placeholder values. The reason we don't have a test for this is that we don't have a good way to test fatal errors in the C++ API.